src/BloomExe/Workspace/BloomClipboard.cs

```diff
@@ -88,15 +118,31 @@ public static PalasoImage GetImageFromClipboard()
 
 			return null;
 #else
-			if (Clipboard.ContainsImage())
-			{
-				return PalasoImage.FromImage(Clipboard.GetImage());
-			}
-
 			var dataObject = Clipboard.GetDataObject();
 			if (dataObject == null)
 				return null;
 
+			var textData = String.Empty;
+			if (dataObject.GetDataPresent(DataFormats.UnicodeText))
+				textData = dataObject.GetData(DataFormats.UnicodeText) as String;
+			if (Clipboard.ContainsImage())
+			{
+				PalasoImage image = null;
+				var haveFileUrl = false;
+				try
+				{
+					image = PalasoImage.FromImage(Clipboard.GetImage()); // this method won't copy any metadata
+					haveFileUrl = !String.IsNullOrEmpty(textData) && File.Exists(textData);
+				}
+				catch (Exception e)
+				{
+					Console.WriteLine(e);
+				}
+				// If we have an image on the clipboard, and we also have text that is a valid url to an image file,
+				// use the url to create a PalasoImage (which will pull in any metadata associated with the image too.
+				return !haveFileUrl ? image : PalasoImage.FromFile(textData);
```

Comment:
The FromFile() call needs to be inside the exception handler, in case the image file is corrupt or even not an image file at all.